pkg/controller/common/rollout/workloads/deployment_controller.go

```diff
@@ -73,15 +73,19 @@ func (c *DeploymentController) VerifySpec(ctx context.Context) (bool, error) {
 	defer func() {
 		if verifyErr != nil {
 			klog.Error(verifyErr)
-			c.recorder.Event(c.parentController, event.Warning("VerifyFailed", verifyErr))
+			if c.recorder != nil && c.parentController != nil {
```

Comment:
when will the recorder be nil?

Reply:
btw, the deployment rollout is not really finished, it doesn't cover many corner cases (i.e. revert in the middle of rollout)

Reply:
@ryanzhang-oss From the normal logic, application rollout plan controller --> rollout plan controller --> deployment controller, the recorder won't be nil.
But from the filed type, it's possible.

>btw, the deployment rollout is not really finished, it doesn't cover many corner cases (i.e. revert in the middle of rollout)

Got it, I will focus on adding unit-tests on Cloneset.